url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/Simulation.java

```diff
@@ -0,0 +1,1042 @@
+							//							cm.setParentId(app_id);
+							//							String cm_id = "lala"; //addEntity(cm, jump);
+							//							cm.setId(cm_id);
+						}
+					}
+				}
+
+				int personcount = instDoc.getPersons().size();
+				// Create the appliances
+				TreeMap<String,Person> existingPersons = new TreeMap<String,Person>();
+				for (int j = 0; j < personcount; j++) {
+					Person personDoc = instDoc.getPersons().get(j);
+					String personid = personDoc.getId();
+					String personName = personDoc.getName();
+					String personDescription = personDoc.getDescription();
+					String personType = personDoc.getType();
+					double awareness = personDoc.getAwareness();
+					double sensitivity = personDoc.getSensitivity();
+					Person person = new Person.Builder(personid, personName, personDescription, personType, inst, awareness, sensitivity).build();
+					Vector<Activity> acts = personDoc.getActivities();
+					int actcount = acts.size();
+					//System.out.println("Act-Count: " + actcount);
+					for (int k = 0; k < actcount; k++) {
+						Activity activityDoc = acts.get(k);
+						String activityName = activityDoc.getName();
+						String activityType = activityDoc.getType();
+						String actid = activityDoc.getId();
+						Activity act = new Activity.Builder(actid, activityName, "", activityType, simulationWorld).build();
+						TreeMap<String, Vector<Appliance>> actModApps = activityDoc.getAppliances();
+						TreeMap<String, Boolean> shiftables = activityDoc.getShiftable();
+						TreeMap<String, Boolean> exclusives = activityDoc.getConfig();
+						TreeMap<String, ProbabilityDistribution> probStartTime = activityDoc.getProbStartTime();
+						TreeMap<String, ProbabilityDistribution> probDuration = activityDoc.getProbDuration();
+						TreeMap<String, ProbabilityDistribution> probeTimes = activityDoc.getnTimesGivenDay();
+						ProbabilityDistribution startDist;
+						ProbabilityDistribution durDist;
+						ProbabilityDistribution timesDist;
+						Vector<Appliance> appliances;
+						for (String key: shiftables.keySet()) {
+							boolean shiftable = shiftables.get(key);
+							boolean exclusive = exclusives.get(key);
+							durDist =  copyProbabilityDistribution(probDuration.get(key), "duration");
+							startDist = copyProbabilityDistribution(probStartTime.get(key), "start");
+							timesDist = copyProbabilityDistribution(probeTimes.get(key), "times");
+							act.addDuration(key, durDist);
+							act.addStartTime(key, startDist);
+							act.addTimes(key, timesDist);
+							act.addShiftable(key, shiftable);
+							act.addConfig(key, exclusive);
+							// add appliances
+							appliances = actModApps.get(key);
+							String[] containsAppliances = new String[appliances.size()];
+							for(int m = 0; m < appliances.size(); m++) 
+								containsAppliances[m] = inst_id + "_" + appliances.get(m).getId();
+							act.addAppliances(containsAppliances, existing, key);
+						}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 379:396, 516:533.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.